File: leet_code/python3/twoSum.py
def twoSum(nums, target):
  possibles = {}
  for idx, num in enumerate(nums):
    if num in possibles:
      return [possibles[num], idx]
    possibles[target - num] = idx

# Time Complexity: O(n)
# Space Complexity: O(n)

nums = [2, 7, 11, 15]
target = 9


print (twoSum(nums, target))

File: leet_code/codesig/twoSum.js
const twoSum = A => {
  let sum = 0;

  for (const num of A) {
    let str = num.toString();
    if (str[0] !== '-' && str.length === 2) sum += num;
    if (str[0] === '-' && str.length === 3) sum += num;
  }

  return sum;
}

let A = [1,1000,80,-91]
console.log(twoSum(A))